src/popup/router/components/RequestAmount.vue

```diff
@@ -0,0 +1,137 @@
+      enough_ae: fee.toString(),
+      not_token: noToken,
+    }"
+    v-bind="$attrs"
+    class="request-amount"
+    type="number"
+    name="amount"
+    placeholder="0.00"
+    new-ui
+    :value="value"
+    :label="$attrs.label || $t('pages.tipPage.amountLabel')"
+    :error="$attrs.error || errors.has('amount')"
+    :error-message="$attrs['error-message'] || errors.first('amount')"
+    @input="$emit('input', $event)"
+  >
+    <template #right>
+      <AssetSelector
+        v-model="selectedAsset"
+        class="request-amount-asset"
+      />
+    </template>
+
+    <template #under="{ focused }">
+      <div
+        class="request-amount-desc"
+        :class="{ focused }"
+      >
+        <span
+          v-if="!selectedToken"
+          class="request-amount-desc-total"
+          data-cy="total-amount-currency"
+        >
+          <span v-if="$attrs.value">&thickapprox;</span>
+          {{ formatCurrency(totalAmount) }}
+        </span>
+
+        <span class="request-amount-desc-at">
+          @{{
+            (currentTokenFiatPrice && selectedAsset && selectedAsset.current_price)
+              ? formatCurrency(currentTokenFiatPrice)
+              : $t('priceNotAvailable')
+          }}
+        </span>
+      </div>
+    </template>
+  </InputField>
+</template>
+
+<script>
+import { mapGetters } from 'vuex';
+import { pick } from 'lodash-es';
+import BigNumber from 'bignumber.js';
+import AssetSelector from './AssetSelector.vue';
+import InputField from './InputField.vue';
+
+export default {
+  components: {
+    AssetSelector,
+    InputField,
+  },
+  props: {
+    value: { type: [String, Number], default: '' },
+    validation: { type: Object, default: () => null },
+    noToken: Boolean,
+    ownValidation: Boolean,
+  },
+  data() {
+    return {
+      fee: BigNumber(0),
+      selectedAsset: null,
+    };
+  },
+  subscriptions() {
+    return pick(this.$store.state.observables, ['balance']);
+  },
+  computed: {
+    ...mapGetters(['formatCurrency', 'account']),
+    ...mapGetters('fungibleTokens', ['selectedToken']),
+    hasError() {
+      return this.$validator.errors.has('amount');
+    },
+    max() {
+      return (this.selectedToken && !this.noToken
+        ? this.selectedToken.balance
+        : this.balance.minus(this.fee)).toString();
+    },
+    totalAmount() {
+      return !this.selectedAsset?.current_price
+        ? 0
+        : ((this.$attrs.value || 0) * (this.selectedAsset.current_price).toFixed(2));
+    },
+    currentTokenFiatPrice() {
+      return `${this.selectedAsset?.current_price?.toFixed(2)}`;
+    },
+  },
+  watch: {
+    async selectedAsset(val) {
+      this.$emit('asset-selected', val);
+    },
+    hasError(value) {
+      return this.$emit('error', value);
+    },
+  },
+};
+</script>
+
+<style lang="scss" scoped>
+@use '../../../styles/variables';
+@use '../../../styles/typography';
+@use '../../../styles/mixins';
+
+.request-amount {
```

Comment:
Missing one small feature: 
    
<img width="529" alt="Screenshot 2022-08-22 at 1 18 45 PM" src="https://user-images.githubusercontent.com/88760043/185868054-e4d8a4c7-740f-48bd-b4f6-ed336ecd2a78.png">

  One of the solution: 
  Within `.request-amount`
  ```scss
  :focus-within .asset-field .asset-selector {
    background-color: #171717;
  }
  ```

Reply:
Asset selector was refactored by Paweł within the scope of a different task: https://aeternity-blockchain.atlassian.net/browse/SW-130

Reply:
Okay. @pstachel knows about this.